### PR TITLE
clientruntime dnxcore50 support

### PR DESCRIPTION
**Pull request by matt-gibbs**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@matt-gibbs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mattgi). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: AutoRest/AutoRest.Core/Logging/CodeGenerationException.cs [hunk 1]
```diff
@@ -9,7 +9,9 @@ namespace Microsoft.Rest.Generator.Logging
     /// <summary>
     /// Code generation aggregate exception.
     /// </summary>
+#if !DNXCORE50
```
Comment (markcowl) [inline AutoRest/AutoRest.Core/Logging/CodeGenerationException.cs]: Let's just remove the serializable attributes for exceptions and add exclusions for any code analysis issues that show up

File: ClientRuntimes/CSharp/ClientRuntime.Azure/CloudException.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace Microsoft.Rest.Azure
     /// <summary>
     /// An exception generated from an http response returned from a Microsoft Azure service
     /// </summary>
-#if !PORTABLE
+#if !PORTABLE && !DNXCORE50
```
Comment (markcowl) [inline ClientRuntimes/CSharp/ClientRuntime.Azure/CloudException.cs]: Same.  Let's remove serializable

File: ClientRuntimes/CSharp/ClientRuntime.Azure/Microsoft.Rest.ClientRuntime.Azure/project.lock.json [hunk 3]
```diff
@@ -0,0 +1,2071 @@
+{
```
Comment (markcowl) [inline ClientRuntimes/CSharp/ClientRuntime.Azure/Microsoft.Rest.ClientRuntime.Azure/project.lock.json]: Is this file necessary, or is it created by the build process?  I assume this is providing the dependency versions for all dependent assemblies

File: ClientRuntimes/CSharp/ClientRuntime/RestException.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ namespace Microsoft.Rest
     /// <summary>
     /// Generic exception for Microsoft Rest Client. 
     /// </summary>
-#if !PORTABLE
+#if !PORTABLE && !DNXCORE50
```
Comment (markcowl) [inline ClientRuntimes/CSharp/ClientRuntime/RestException.cs]: same comment

**Comment by devigned** [comment]
@matt-gibbs thank you for getting the build back to green!

File: ClientRuntimes/CSharp/ClientRuntime.Azure/Microsoft.Rest.ClientRuntime.Azure/project.lock.json (hunk 3 above)
Comment (matt-gibbs) [inline ClientRuntimes/CSharp/ClientRuntime.Azure/Microsoft.Rest.ClientRuntime.Azure/project.lock.json]: It is "locking" the dependencies calculated at that time. dnu build uses what is in the project.lock.json

File: AutoRest/AutoRest.Core/Logging/CodeGenerationException.cs (hunk 1 above)
Comment (matt-gibbs) [inline AutoRest/AutoRest.Core/Logging/CodeGenerationException.cs]: ok with doing that as a follow-up PR?

**Comment by devigned** [comment]
@matt-gibbs and @markcowl you guys can figure out what happens with the attributes. I'm not sure if they should be removed, but that would be a good one to talk over (see: https://www.pivotaltracker.com/story/show/103157700).
